google/cloud/storage/benchmarks/storage_shard_throughput_benchmark.cc

```diff
@@ -0,0 +1,481 @@
+    }
+  }
+  if (!group.empty()) {
+    tasks.emplace_back(
+        std::async(std::launch::async, &DeleteGroup, client, std::move(group)));
+  }
+  int count = 0;
+  for (auto& t : tasks) {
+    auto status = t.get();
+    if (!status.ok()) {
+      std::cerr << "Error return task[" << count << "]: " << status << "\n";
+    }
+    ++count;
+  }
+  // We do not print the latency to delete the objects because we have another
+  // benchmark to measure that.
+  auto elapsed = std::chrono::steady_clock::now() - start;
+  std::cout << "# Deleted in " << duration_cast<milliseconds>(elapsed).count()
+            << "ms\n";
+}
+
+google::cloud::StatusOr<Options> ParseArgs(int argc, char* argv[]) {
+  Options options;
+  bool wants_help = false;
+  bool wants_description = false;
+  std::vector<gcs_bm::OptionDescriptor> desc{
+      {"--help", "print usage information",
+       [&wants_help](std::string const&) { wants_help = true; }},
+      {"--description", "print benchmark description",
+       [&wants_description](std::string const&) { wants_description = true; }},
+      {"--project-id", "use the given project id for the benchmark",
+       [&options](std::string const& val) { options.project_id = val; }},
+      {"--bucket-prefix", "configure the bucket's prefix",
+       [&options](std::string const& val) { options.bucket_prefix = val; }},
+      {"--region", "use the given region for the benchmark",
+       [&options](std::string const& val) { options.region = val; }},
+      {"--object-count", "set the number of objects created by the benchmark",
+       [&options](std::string const& val) {
+         options.object_count = std::stoi(val);
+       }},
+      {"--thread-count", "set the number of threads in the benchmark",
+       [&options](std::string const& val) {
+         options.thread_count = std::stoi(val);
+       }},
+      {"--iteration-size",
+       "set the number of chunk downloaded in each iteration",
+       [&options](std::string const& val) {
+         options.iteration_size = std::stoi(val);
+       }},
+      {"--iteration-count",
+       "set the number of samples captured by the benchmark",
+       [&options](std::string const& val) {
+         options.iteration_count = std::stoi(val);
+       }},
+      {"--chunk-size", "size of the chunks used in the benchmark",
+       [&options](std::string const& val) {
+         options.chunk_size = gcs_bm::ParseSize(val);
+       }},
+      {"--chunk-count", "the number of chunks in each object",
+       [&options](std::string const& val) {
+         options.chunk_count = std::stoi(val);
+       }},
+  };
+  auto usage = gcs_bm::BuildUsage(desc, argv[0]);
+
+  auto unparsed = gcs_bm::OptionsParse(desc, {argv, argv + argc});
+  if (wants_help) {
+    std::cout << usage << "\n";
+  }
+
+  if (wants_description) {
+    std::cout << kDescription << "\n";
+  }
+
+  if (unparsed.size() > 2) {
+    std::ostringstream os;
+    os << "Unknown arguments or options\n" << usage << "\n";
```

Comment:
Non-actionable observation: Newlines in status messages are a little odd, particularly terminating ones.

Reply:
Ack.